mitodl/mit-open
# Basic learning resources drawer

### What are the relevant tickets?
#642 

### Description (What does it do?)
This PR adds a basic learning resource drawer:
- minimal resource metadata for now
- renders playable video for video, image for everything else
- opens by clicking title of LearningResourceCard

Also:
- Moves `EmbedlyCard` to the `ol-components` package. This seems like a better place for it. This is why so many files are touched.

### Screenshots (if appropriate):

**Note:** Resource descriptions can contain HTML strings. Currently, I'm rendering this as a string. If we want to render as HTML (not sure we do) we can address that in a followup PR.

<img width="1372" alt="Screenshot 2024-03-29 at 4 22 13 PM" src="https://github.com/mitodl/mit-open/assets/9010790/7c98c22f-d1e7-4fe7-8f13-fea6065469f0">

### How can this be tested:

1. View the "ExpandedLearningResourceDetails" story.
    - `docker compose up`
    - `docker compose exec watch yarn storybook`
    - View the last "story" on left hand sidebar, "ExpandedLearningResourceDetails". This shows roughly what the card looks lik. There's a dropdown where you can pick different resource types. Right now only "Video" is different from the others.
2. Go to `htpp://localhost:8063`. Or view the searchpage.  Any learning resource card should have a clickable title that opens the resource drawer.

Some finer points to look for:
- When the drawer content is loading, it should show a skeleton UI. 
- The drawer content should be visible even while the drawer is opening/closing. (This is a little hard to see unless you add something like `transitionDuration={2000}` to the RoutedDrawer in LearningResourceDrawer.tsx
